src/calendar/calendar.tsx

```diff
@@ -0,0 +1,394 @@
+
+        if (isNextMonthSelected) setCalendarDate(dayjs(value));
+
+        setSelectedStartDate(value);
+
+        performOnChangeHandler(value);
+    };
+
+    const handleMonthYearClick = (
+        e: React.MouseEvent<HTMLDivElement>,
+        show: View
+    ) => {
+        const value = e.currentTarget.getAttribute("data-value") || "";
+        let isYearDecaded = false;
+        if (show === "Year") {
+            isYearDecaded =
+                e.currentTarget.getAttribute("data-variant") === "nextDecaded"
+                    ? true
+                    : false;
+        }
+
+        if (!value.length) return;
+        const hasStartDate: string[] = selectedStartDate.split("-");
+
+        const [yyyy, mm, _dd] = hasStartDate;
+        const selectedMonth = value.split("-")[1];
+        const selectedYear = value.split("-")[0];
+
+        if (hasStartDate.length === 3) {
+            /**
+             * Handle last day check/validate
+             * user select 2023-01-30 and select month calendar which didn't exist 30 day in the month
+             * it will get the last day of the month
+             */
+            const lastDayInMonth: number = dayjs(
+                `${yyyy}-${selectedMonth}-01`
+            ).daysInMonth();
+            let dd: string = _dd;
+
+            if (+_dd > lastDayInMonth) {
+                dd = lastDayInMonth.toString();
+            }
+
+            // date already been selected
+            const fullDate =
+                show === "Month"
+                    ? `${selectedYear}-${selectedMonth}-${dd}`
+                    : showView === "Year"
+                    ? `${selectedYear}-${mm}-${dd}`
+                    : dayjs().format("YYYY-MM-DD");
+
+            if (isYearDecaded) setCalendarDate(dayjs(value));
+            setSelectedStartDate(fullDate);
+            return;
+        }
+
+        // wihtout selected date yet
+        if (isYearDecaded) setCalendarDate(dayjs(value));
+        setSelectedStartDate(value);
+    };
+
+    // =============================================================================
+    // Calendar Generate
+    // =============================================================================
+    const firstDayOfTheMonth = useMemo(
+        () => calendarDate.clone().startOf("month"),
+        [calendarDate]
+    );
+
+    const firstDayOfFirstWeekOfMonth = useMemo(
+        () => dayjs(firstDayOfTheMonth).startOf("week"),
+        [firstDayOfTheMonth]
+    );
+
+    const generateFirstDayOfEachWeek = useCallback((day: Dayjs): Dayjs[] => {
+        const dates: Dayjs[] = [day];
+        for (let i = 1; i < 6; i++) {
+            const date = day.clone().add(i, "week");
+            dates.push(date);
+        }
+        return dates;
+    }, []);
+
+    const generateWeek = useCallback((day: Dayjs): Date[] => {
+        const dates: Date[] = [];
+        for (let i = 0; i < 7; i++) {
+            const date = day.clone().add(i, "day").toDate();
```

Comment:
so far the component has been manipulating the dates with `Dayjs`, so why convert to `Date` here?

Reply:
`Dayjs` was a object from the dayJs package. In line 258, `toDate()` will be formatted to `Fri Feb 03 2023 13:53:27 GMT+0800 (Malaysia Time)` Date String instead of DayJs object

Reply:
Handle a Raw Date format is easy to readable and do other function from the raw Date format. 

The DayJs is an object from the package. Need to be transform back to raw format in other function.